File: Collector/OLD/LegacyJava/Collector/Character/Mouse.java
package Collector.Character;

import Collector.Restrictions;
import com.badlogic.gdx.graphics.Texture;
import com.badlogic.gdx.math.Vector3;

public class Mouse implements Restrictions {
    Texture crosshair;

    public Mouse() {
        this.crosshair = new Texture("assets/crosshair.png");
    }

    public Texture getCrosshair() {
        return crosshair;
    }

    public int getSelectedX(Vector3 mousePos) {
        return ((int)(mousePos.x)>>4)<<4;
    }

    public int getSelectedY(Vector3 mousePos) {
        return ((int)(mousePos.y)>>4)<<4;
    }
}

File: Collector/src/Character/Mouse.cs
using Microsoft.Xna.Framework;
using Microsoft.Xna.Framework.Content;
using Microsoft.Xna.Framework.Graphics;
using Microsoft.Xna.Framework.Input;
using MonoGame.Extended;

namespace Collector.Character
{
    public class Mouse : IRestrictions
    {
        private SpriteBatch _spriteBatch;
        private ContentManager _contentManager;
        private OrthographicCamera _cam;
        private Texture2D Crosshair { get; set; }

        public Mouse(ContentManager contentManager, SpriteBatch spriteBatch, OrthographicCamera cam)
        {
            _spriteBatch = spriteBatch;
            _cam = cam;
            _contentManager = contentManager;
            Crosshair = _contentManager.Load<Texture2D>("crosshair");
        }

        public void Draw()
        {
            _spriteBatch.Draw(
                Crosshair,
                new Vector2(
                    GetSelectedX(Microsoft.Xna.Framework.Input.Mouse.GetState().X-(IRestrictions.TileSize/2f)) + (_cam.Position.X)+(IRestrictions.TileSize/2f), 
                    GetSelectedY(Microsoft.Xna.Framework.Input.Mouse.GetState().Y-(IRestrictions.TileSize/2f)) + (_cam.Position.Y)+(IRestrictions.TileSize/2f)
                    ),
                Color.White
            );
        }


        private static int GetSelectedX(float x)
        {
            return ((int) x >> 5) << 5;
        }

        private static int GetSelectedY(float y)
        {
            return ((int) y >> 5) << 5;
        }
    }
}